src/tests/system/tests/test_gdm.py

```diff
@@ -0,0 +1,204 @@
+        3. Configure SSSD pam_json_services = gdm-switchable-auth
+    :steps:
+        1. Select user from list (if not listed, use Other and enter user)
+        2. Follow IdP redirect url and login as user via External IdP site
+        3. Check logs for login denial
+    :expectedresults:
+        1. User selected and login method (EIdP) chosen
+        2. User connects to remote URL with browser and sees login is disabled
+        3. Should see login denial in logs
+    :customerscenario: False
+    """
+    testuser = "kcgdmuser1"
+    password = "Secret123"
+    testuser_idp = f"{testuser}@{keycloak.host.hostname}"
+
+    client.sssd.import_domain("gdm.test", ipa)
+    client.sssd.pam["pam_json_services"] = "gdm-switchable-auth"
+    client.sssd.start()
+
+    keycloak.user(testuser).add()
+    keycloak.user(testuser).modify(email=f"{testuser_idp}")
+    ipa.user(testuser).add()
+    ipa.user(testuser).modify(user_auth_type="idp", idp="keycloak", idp_user_id=testuser_idp)
+    keycloak.user(testuser).modify(enabled=False)
+
+    assert not client.gdm.login_idp(client, testuser, password), "GDM EIdP Login passed when it should have failed!"
+
+
+@pytest.mark.topology(KnownTopology.GDM)
+@pytest.mark.builtwith(client="gdm")
+def test_gdm__xidp_login_password_change(client: Client, ipa: IPA, keycloak: Keycloak):
+    """
+    :title: Login via GDM with external IdP user is rejected when login is disabled
+    :setup:
+        1. Configure IPA for External IdP support
+        2. Add user to IPA as IdP user with authtype only set to 'idp'
+        3. Configure SSSD pam_json_services = gdm-switchable-auth
+    :steps:
+        1. Select user from list (if not listed, use Other and enter user)
+        2. Follow IdP redirect url and login as user via External IdP site
+        3. Check logs for login denial
+    :expectedresults:
+        1. User selected and login method (EIdP) chosen
+        2. User connects to remote URL with browser and sees login is disabled
+        3. Should see login denial in logs
+    :customerscenario: False
+    """
+    testuser = "kcgdmuser1"
+    password = "Secret123"
+    testuser_idp = f"{testuser}@{keycloak.host.hostname}"
+
+    client.sssd.import_domain("gdm.test", ipa)
+    client.sssd.pam["pam_json_services"] = "gdm-switchable-auth"
+    client.sssd.start()
+
+    keycloak.user(testuser).add()
+    keycloak.user(testuser).modify(email=f"{testuser_idp}")
+    ipa.user(testuser).add()
+    ipa.user(testuser).modify(user_auth_type="idp", idp="keycloak", idp_user_id=testuser_idp)
+
+    keycloak.user(testuser).password_change_at_logon()
+
+    assert client.gdm.login_idp(
+        client, testuser, f"{password}:::NewPa55"
+    ), "GDM EIdP Login with password change failed!"
+
+
+@pytest.mark.topology(KnownTopology.GDM)
+@pytest.mark.builtwith(client="gdm")
+def test_gdm__xidp_login_get_kerberos_ticket(client: Client, ipa: IPA, keycloak: Keycloak):
```

Comment:
Can we merger this test case with `test_gdm__xidp_login`. They are doing almost the same

Reply:
I think I wanted to see in a report when login worked but kerberos didn't but, we can still get that from this single test case by drilling down to the specific failure.   So, I think we can drop the first test case entirely and keep this one.   Sound reasonable?

Reply:
Yes, that sounds good